test/server/cards/01_SOR/events/SneakAttack.spec.ts

```diff
@@ -0,0 +1,47 @@
+describe('Sneak Attack', function() {
+    integration(function(contextRef) {
+        describe('Sneak Attack\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['sneak-attack', 'sabine-wren#you-can-count-on-me', 'obiwan-kenobi#following-fate', 'recruit'],
+                        groundArena: ['battlefield-marine'],
+                        spaceArena: ['cartel-spacer'],
+                        base: 'administrators-tower',
+                        leader: 'luke-skywalker#faithful-friend',
+                        resources: 3
+                    },
+                    player2: {
+                        groundArena: ['atst'],
+                    }
+                });
+            });
+
+            it('should play Sabine for 1 resource less and defeat it at the end.', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.sneakAttack);
+                expect(context.player1).toBeAbleToSelectExactly([context.sabineWren]);
+                context.player1.clickCard(context.sabineWren);
+                expect(context.sabineWren.exhausted).toBeFalse();
+                expect(context.player1.readyResourceCount).toBe(0);
+
+                // Check that Sabine is defeated at the beginning of the regroup phase
+                context.moveToRegroupPhase();
+                expect(context.sabineWren).toBeInZone('discard');
+                expect(context.player1).toHavePrompt('Select between 0 and 1 cards to resource');
+            });
+
+            it('should not bug if Sabine is defeated before the end of the phase', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.sneakAttack);
+                context.player1.clickCard(context.sabineWren);
+                context.player2.clickCard(context.atst);
+                context.player2.clickCard(context.sabineWren);
+                expect(context.sabineWren).toBeInZone('discard');
+            });
+        });
```

Comment:
Apologies, one other test I realized we need to check a potential bug - could you add a test where we play out the Sabine, then waylay her back to hand and play her again? In that situation, per the swu rules, she _shouldn't_ be defeated by the delayed effect since she's technically a new unit.

I'm not 100% sure that is implemented correctly in the engine though. If it doesn't work right then just comment it out with a TODO and I'll add a work item for fixing that

Reply:
She is indeed defeated at the end. I'll comment the test and add the TODO.